hack/main.go

```diff
@@ -140,37 +101,83 @@ func CreateEC2Instance(ec2Client *ec2.Client, amiID, instanceType string, instan
 
 	return *instanceResp, nil
 }
-func DescribeEC2Instances( client *ec2.Client, instanceID string) (int, error) {
-     // States and codes
-     // 0 : pending
-     // 16 : running
-     // 32 : shutting-down
-     // 48 : terminated
-     // 64 : stopping
-     // 80 : stopped
-     // 401 : failed
-     result, err := client.DescribeInstanceStatus(context.TODO(), &ec2.DescribeInstanceStatusInput{
-         InstanceIds: []string{instanceID},
-     })
- 
-     if err != nil {
-         panic(fmt.Sprintf("Errors while describing the instance status: %s\n", err.Error()))
-         if aerr, ok := err.(awserr.Error); ok {
-             if aerr.Code() == "UnauthorizedOperation" {
-                 return 401, err
-             }
-         }
-         return 0, err
-     }
- 
-     if len(result.InstanceStatuses) > 1 {
-         return 0, errors.New("more than one EC2 instance found")
-     }
- 
-     if len(result.InstanceStatuses) == 0 {
-         return 0, errors.New("no EC2 instances found")
-     }
-     return int(*result.InstanceStatuses[0].InstanceState.Code), nil
+
+// Returns state code as int
+func DescribeEC2Instances(client *ec2.Client, instanceID string) (int, error) {
+	// States and codes
+	// 0 : pending
+	// 16 : running
+	// 32 : shutting-down
+	// 48 : terminated
+	// 64 : stopping
+	// 80 : stopped
+	// 401 : failed
+	result, err := client.DescribeInstanceStatus(context.TODO(), &ec2.DescribeInstanceStatusInput{
+		InstanceIds: []string{instanceID},
+	})
+
+	if err != nil {
+		panic(fmt.Sprintf("Errors while describing the instance status: %s\n", err.Error()))
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == "UnauthorizedOperation" {
+				return 401, err
+			}
+		}
+		return 0, err
+	}
+
+	if len(result.InstanceStatuses) > 1 {
+		return 0, errors.New("more than one EC2 instance found")
+	}
+
+	if len(result.InstanceStatuses) == 0 {
+		// Don't return an error here as if the instance is still too new, it may not be
+		// retured at all.
+		//return 0, errors.New("no EC2 instances found")
```

Comment:
@lnguyen1401 I had to comment out this line here, as once I changed the function to properly check the error within the loop, this error caused a panic. It looks like AWS doesn't return an instance status at all if the instance is still very new, so the first couple times describe is called, its likely that 0 instances are returned.

Reply:
Ohh i see what you mean!